libsolidity/analysis/ImmutableValidator.cpp

```diff
@@ -104,17 +105,47 @@ bool ImmutableValidator::visit(MemberAccess const& _memberAccess)
 
 bool ImmutableValidator::visit(IfStatement const& _ifStatement)
 {
+	// store old context variables
 	bool prevInBranch = m_inBranch;
+	VariableDeclarationSet prevWrites = std::move(m_immutableStateVariablesWrittenInBranch);
 
 	_ifStatement.condition().accept(*this);
 
 	m_inBranch = true;
+	m_immutableStateVariablesWrittenInBranch.clear();
 	_ifStatement.trueStatement().accept(*this);
 
+	// find out immutable state variables that are written in then/else branches
+	auto thenWrites = std::move(m_immutableStateVariablesWrittenInBranch);
+	VariableDeclarationSet elseWrites;
 	if (auto falseStatement = _ifStatement.falseStatement())
+	{
+		m_immutableStateVariablesWrittenInBranch.clear();
 		falseStatement->accept(*this);
+		elseWrites = std::move(m_immutableStateVariablesWrittenInBranch);
+	}
 
+	// restore and update context
 	m_inBranch = prevInBranch;
+	m_immutableStateVariablesWrittenInBranch = std::move(prevWrites);
+	std::vector<ASTNode::SourceLocation const*> bad_write_locs;
```

Comment:
```suggestion
	std::vector<ASTNode::SourceLocation const*> badWriteLocations;
```
(style guide also says "no abbreviations", or at least keep them to a minimum)

Reply:
Oh it's actually not even used?

Reply:
My bad -- left from old commit. Removed.